File: LeetCodes/single-number-iii/7-4-2023, 5_38_04 PM/single-number-iii.cpp
// Link: https://leetcode.com/problems/single-number-iii

/** Submission Info:
 *
 *  AUTHOR: the_247 | Raj Patel
 *  TITLE: single-number-iii
 *  RUNTIME: 7 ms
 *  MEMORY: 10.7 MB
 *  DATE: 7-4-2023, 5:38:04 PM
 *
 */

// Solution:

class Solution {
public:
    vector<int> singleNumber(vector<int>& nums) {
        unordered_map<int,int> cnt;
        vector<int> ans;
        for(auto &val: nums) {
            cnt[val]++;
        }
        for(auto &x: cnt) {
            if(x.second == 1) {
                ans.push_back(x.first);
            }
        }
        return ans;
    }
};

File: LeetCodes/single-number-iii/5-31-2024, 12_17_58 PM/single-number-iii.java
// Link: https://leetcode.com/problems/single-number-iii

/** Submission Info:
 *
 *  AUTHOR: the_247 | Raj Patel
 *  TITLE: single-number-iii
 *  RUNTIME: 7 ms
 *  MEMORY: 45.3 MB
 *  DATE: 5-31-2024, 12:17:58 PM
 *
 */

// Solution:

class Solution {
    public int[] singleNumber(int[] nums) {
        Map<Integer, Integer> frequencyMap = new HashMap<>();
        
        for (int num : nums) {
            frequencyMap.put(num, frequencyMap.getOrDefault(num, 0) + 1);
        }
        
        int[] result = new int[2];
        int index = 0;
        for (Map.Entry<Integer, Integer> entry : frequencyMap.entrySet()) {
            if (entry.getValue() == 1) {
                result[index++] = entry.getKey();
            }
        }
        
        return result;
    }
}
